extends Node2D


const _DIALOG_SCREEN : PackedScene = preload("res://dialog/dialog_screen.tscn")

var _dialog_data1 : Dictionary = {
	0: {
		"faceset": "res://sprites/faceset/DanceCreature.jpeg",
		"dialog": "Ugh! Eu odeio isso! Odeio esse lugar! Odeio dan√ßar! Por que eu sou o √∫nico que n√£o consegue?",
		"title": "StumbleFeet",
	},
	1: {
		"faceset": "res://sprites/faceset/HopeBringer.png",
		"dialog": "üëè",
		"title": "Hope Bringer",
	},
	2: {
		"faceset": "res://sprites/faceset/DanceCreature.jpeg",
		"dialog": "O que voc√™ quer? Vai me dizer que eu preciso dan√ßar tamb√©m? Vai rir de mim como todos os outros?",
		"title": "StumbleFeet",
	},
	3: {
		"faceset": "res://sprites/faceset/HopeBringer.png",
		"dialog": "ÔΩû(‚ñ†_‚ñ†)ÔΩû‚ô™üëâ",
		"title": "Hope Bringer",
	},
	4: {
		"faceset": "res://sprites/faceset/DanceCreature.jpeg",
		"dialog": "Voc√™ acha que pode me ensinar a arte do gingado? Logo eu que n√£o tenho nem rebolado.",
		"title": "StumbleFeet",
	},
	5: {
		"faceset": "res://sprites/faceset/HopeBringer.png",
		"dialog": "ü•∫üëâüëà",
		"title": "Hope Bringer",
		},
	6: {
		"faceset": "res://sprites/faceset/DanceCreature.jpeg",
		"dialog": "T√° bom, t√° bom. Mostra a√≠, eu vou tentar.",
		"title": "StumbleFeet",
		},
}

@export_category("Object")
@export var hud : CanvasLayer = null

# Called when the node enters the scene tree for the first time.
func _ready():
	TransitionScreen._fade_out()


func _on_area_2d_body_entered(body):
	if body.is_in_group("player"):
		var _new_dialog:DialogScreen = _DIALOG_SCREEN.instantiate()
		_new_dialog.data = _dialog_data1
		hud.add_child(_new_dialog)
		$Player.on_dialog = true
		
		await _new_dialog.dialog_ended
		
		$Player.on_dialog = false
		if body.is_in_group("player"):
			TransitionScreen._fade_transition()
			await get_tree().create_timer(1).timeout
			get_tree().change_scene_to_file("res://scenes/second_level/cave_level/dance_game1.tscn")




# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	pass

func _on_visible_on_screen_notifier_2d_screen_exited():
	$Player.position = Vector2(32, -5)

